jdaviz/configs/cubeviz/plugins/tools.py

```diff
@@ -42,3 +47,133 @@ def on_mouse_event(self, data):
         self.viewer.session.hub.broadcast(msg)
 
         self._time_last = time.time()
+
+
+@viewer_tool
+class SpectrumPerSpaxel(CheckableTool):
+
+    icon = os.path.join(ICON_DIR, 'pixelspectra.svg')
+    tool_id = 'jdaviz:spectrumperspaxel'
+    action_text = 'See spectrum at a single spaxel'
+    tool_tip = 'Click on the viewer and see the spectrum at that spaxel in the spectrum viewer'
+
+    def __init__(self, viewer, **kwargs):
+        self.old_y_min = 0
+        self.old_y_max = 0
+        self.spectrum_viewer = None
+
+        # Keep track of data created during this session and remove
+        # them from the spectrum viewer after the tool is deactivated
+        self.created_data_labels = []
+
+        # #b515d1 is magenta
+        self.colors = ['#b515d1']
+        super().__init__(viewer, **kwargs)
+
+    def activate(self):
+        self.viewer.add_event_callback(self.on_mouse_event,
+                                       events=['click'])
+        self.spectrum_viewer = self.viewer.jdaviz_app.get_viewer_by_id("cubeviz-3")
+
+    def deactivate(self):
+        self.viewer.remove_event_callback(self.on_mouse_event)
+        self.remove_highlight_on_pixel()
+        for label in self.created_data_labels:
+            self.viewer.jdaviz_app.remove_data_from_viewer("spectrum-viewer", label)
+        self.created_data_labels = []
+
+    def on_mouse_event(self, data):
+        event = data['event']
+
+        if event == 'click':
```

Comment:
I've never had to watch for mouse events before, but this feels heavy handed (this method is going to fire for all mouse events then check to see if it's a click event). Is there a way to monitor for ONLY click events? Trying to think about performance

Reply:
Yes, that is basically what it is doing under `activate` method. The `if` is still good to have in case we want to expand this logic to other mouse/key events.

```python
        self.viewer.add_event_callback(self.on_mouse_event,
                                       events=['click'])
```